tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { KubernetesCommandLineToolsExecutor } from '../../utils/KubernetesCommandLineToolsExecutor';
+import { expect } from 'chai';
+import { ShellString } from 'shelljs';
+import { GitUtil } from '../../utils/vsc/GitUtil';
+import { TestConstants } from '../../constants/TestConstants';
+import { DevWorkspaceConfigurationHelper } from '../../utils/DevWorkspaceConfigurationHelper';
+import { DevfileContext } from '@eclipse-che/che-devworkspace-generator/lib/api/devfile-context';
+
+suite(`Empty workspace API test`, async function (): Promise<void> {
+    // works only for root user
+    const namespace: string = TestConstants.TS_API_TEST_NAMESPACE;
+    let clonedProjectName: string;
+    let containerWorkDir: string;
+    let devWorkspaceConfigurationHelper: DevWorkspaceConfigurationHelper;
+    let kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor;
+    let devfileContext: DevfileContext;
+    let devWorkspaceName: string | undefined;
+    let containerTerminal: KubernetesCommandLineToolsExecutor.ContainerTerminal;
+
+    const gitRepository: string = 'https://github.com/crw-qe/web-nodejs-sample';
+
+    suiteSetup('Create empty workspace with OC client', async function (): Promise<void> {
+        const workspaceName: string = 'empty-' + Math.floor(Math.random() * 1000);
+        const devfileContent: string =
+            'schemaVersion: 2.2.0\n' +
+            'metadata:\n' +
+            `  name: ${workspaceName}\n`;
+
+        devWorkspaceConfigurationHelper = new DevWorkspaceConfigurationHelper({
+            devfileContent,
+        });
+        devfileContext = await devWorkspaceConfigurationHelper.generateDevfileContext();
+        devWorkspaceName = devfileContext?.devWorkspace?.metadata?.name;
+        kubernetesCommandLineToolsExecutor = new KubernetesCommandLineToolsExecutor(devWorkspaceName, namespace);
+        containerTerminal = new KubernetesCommandLineToolsExecutor.ContainerTerminal(kubernetesCommandLineToolsExecutor);
+    });
+
+    test('Create empty workspace', async function (): Promise<void> {
+        kubernetesCommandLineToolsExecutor.loginToOcp();
+        const devWorkspaceConfigurationYamlString: string = await devWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString(devfileContext);
+        const output: ShellString = kubernetesCommandLineToolsExecutor.applyAndWaitDevWorkspace(devWorkspaceConfigurationYamlString);
+        expect(output.stdout).contains('condition met');
+    });
+
+    suiteTeardown('Delete workspace', function (): void {
+        kubernetesCommandLineToolsExecutor.deleteDevWorkspace();
+    });
+
+    suite('Clone public repo without previous setup', function (): void {
+        suiteTeardown('Delete cloned project', function (): void {
```

Comment:
tear-down steps are usually placed at the end of test, for the sake of readiness